packages/@lwc/integration-tests/src/components/memory-leaks/test-component-leak/integration/component-leak/component-leak.js

```diff
@@ -0,0 +1,13 @@
+import { LightningElement, createElement } from 'lwc';
+import Child from '../child/child.js';
+
+export default class extends LightningElement {
+    addChild() {
+        this.refs.container.appendChild(createElement('integration-child', { is: Child }));
+    }
+
+    removeChildren() {
+        // Our monkey-patching for synthetic lifecycle events covers removeChild, but not innerHTML = ''
+        this.refs.container.innerHTML = '';
```

Comment:
This is the core of the bug here. If we were using native lifecycle hooks, we wouldn't have this problem.